### PR TITLE
Temp Certificate Bug Fixes; --lonely Ursula

### PR DESCRIPTION
**Pull request by KPrasch**
Based on #659

### CODE REVIEW

File: nucypher/cli/main.py [hunk 1]
```diff
@@ -146,13 +144,15 @@ def status(click_config, config_file):
 @click.option('--quiet', '-Q', help="Disable logging", is_flag=True)
 @click.option('--dry-run', '-x', help="Execute normally without actually starting the node", is_flag=True)
 @click.option('--force', help="Don't ask for confirmation", is_flag=True)
+@click.option('--lonely', help="Do not connect to seednodes", is_flag=True)
```
Comment (tuxxy) [inline nucypher/cli/main.py]: :+1: neat!

File: nucypher/config/storages.py [hunk 2]
```diff
@@ -343,6 +330,12 @@ def __write_tls_certificate(self,
         if not host:
             host = common_name_on_certificate
 
+        pseudonym = certificate.subject.get_attributes_for_oid(NameOID.PSEUDONYM)[0]
```
Comment (tuxxy) [inline nucypher/config/storages.py]: :+1:

**Comment by jMyles** [comment]
This is good stuff.

My qualm is that you have nixed the single demo ursula script (which was useful for examining one Ursula among several in the fleet, in a debugger), while replacing it with the lonely Ursula, which, if I understand correctly, won't even connect to the rest of the fleet (at least, via seed nodes).

So this breaks what has become my normal workflow.

**Comment by KPrasch** [comment]
This an be done by running `nucypher ursula run --dev --teacher-uri localhost:11501`; Which may be minimal enough to not require a script. What do you think?

File: examples/run_demo_ursula_fleet.py [hunk 3]
```diff
@@ -29,44 +29,55 @@
...
-    ursulas, ursula_processes = set(), list()
+
+    ursula_processes = list()
     for index, port in enumerate(ports):
 
-        executable = 'nucypher'
-        args = ['nucypher', 'ursula', 'run',
-                '--federated-only', '--rest-port', port,
-                '--dev', '--debug']
+        teacher_uri = '127.0.0.1:{}'.format(int(port) - 1)
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply examples/run_demo_ursula_fleet.py]: Yes - Although - I have had better success with a common teacher, lonely Ursula.  I think I will switch it back.

[the remark this answers is not in the document]
Comment (KPrasch) [reply examples/run_demo_ursula_fleet.py]: Actually - looks like I changed this in the next PR in `58b337a`.  Perhaps I'll rearrange these?

**Comment by jMyles** [comment]
> It's True that the intention of the lonely Ursula script is to behave as a bootnode (perhaps can be renamed).
> 
> I surmise your needs can be met by running `nucypher ursula run --dev --teacher-uri localhost:11501`; Which may be minimal enough to not require a script. What do you think?

Well, the thing is, `run_single_demo_ursula` performed exactly that function for me, but in a python script that made sense to run in a debugger.

When you want to run an Ursula in the pycharm debugger, what steps do you take?

File: examples/run_demo_ursula_fleet.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (KPrasch) [reply examples/run_demo_ursula_fleet.py]: :heavy_check_mark: - Cherry picked the commit in question

**Comment by KPrasch** [comment]
@jMyles - Yeah - Good Points.  Restored the entry point for a single demo Ursula. :heavy_check_mark:

File: nucypher/network/nodes.py [hunk 4]
```diff
@@ -454,7 +465,9 @@ def cycle_teacher_node(self):
         # that we have connected to all the seed nodes.
         if self.unresponsive_seed_nodes:
             self.log.info("Still have unresponsive seed nodes; trying again to connect.")
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply nucypher/network/nodes.py]: :heavy_check_mark:

File: nucypher/config/storages.py [hunk 5]
```diff
@@ -178,51 +179,39 @@ def get(self,
...
-                               ) -> str:
+        if not is_checksum_address(checksum_address):
+            raise RuntimeError("Invalid certificate checksum_address encountered")  # TODO: More
 
         self.__certificates[checksum_address] = certificate
+
+        certificate_bytes = certificate.public_bytes(self.TLS_CERTIFICATE_ENCODING)
+
+        filename = '{}.pem'.format(checksum_address)
+        filepath = os.path.join(self.__temp_certificates_dir, filename)
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply nucypher/config/storages.py]: :heavy_check_mark:  - Fixed in `69bc9d4`

**Pull request merged**